fontforgeexe/cvpalettes.c

```diff
@@ -3518,34 +3518,72 @@ void CVToolsPopup(CharView *cv, GEvent *event) {
 
     if( anysel )
     {
-	    mi[i].ti.text = (unichar_t *) _("Curve");
-	    mi[i].ti.text_is_1byte = true;
-	    mi[i].ti.fg = COLOR_DEFAULT;
-	    mi[i].ti.bg = COLOR_DEFAULT;
-	    mi[i].mid = MID_Curve;
-	    mi[i].invoke = CVMenuPointType;
-	    i++;
-	    mi[i].ti.text = (unichar_t *) _("HVCurve");
-	    mi[i].ti.text_is_1byte = true;
-	    mi[i].ti.fg = COLOR_DEFAULT;
-	    mi[i].ti.bg = COLOR_DEFAULT;
-	    mi[i].mid = MID_HVCurve;
-	    mi[i].invoke = CVMenuPointType;
-	    i++;
-	    mi[i].ti.text = (unichar_t *) _("Corner");
-	    mi[i].ti.text_is_1byte = true;
-	    mi[i].ti.fg = COLOR_DEFAULT;
-	    mi[i].ti.bg = COLOR_DEFAULT;
-	    mi[i].mid = MID_Corner;
-	    mi[i].invoke = CVMenuPointType;
-	    i++;
-	    mi[i].ti.text = (unichar_t *) _("Tangent");
-	    mi[i].ti.text_is_1byte = true;
-	    mi[i].ti.fg = COLOR_DEFAULT;
-	    mi[i].ti.bg = COLOR_DEFAULT;
-	    mi[i].mid = MID_Tangent;
-	    mi[i].invoke = CVMenuPointType;
-	    i++;
+        if ( cv->b.sc->inspiro && hasspiro()) {
+            mi[i].ti.text = (unichar_t *) _("G4 Curve");
+            mi[i].ti.text_is_1byte = true;
+            mi[i].ti.fg = COLOR_DEFAULT;
+            mi[i].ti.bg = COLOR_DEFAULT;
+            mi[i].mid = MID_SpiroG4;
+            mi[i].invoke = CVMenuPointType;
+            i++;
+            mi[i].ti.text = (unichar_t *) _("G2 Curve");
+            mi[i].ti.text_is_1byte = true;
+            mi[i].ti.fg = COLOR_DEFAULT;
+            mi[i].ti.bg = COLOR_DEFAULT;
+            mi[i].mid = MID_SpiroG2;
+            mi[i].invoke = CVMenuPointType;
+            i++;
+            mi[i].ti.text = (unichar_t *) _("Corner");
+            mi[i].ti.text_is_1byte = true;
+            mi[i].ti.fg = COLOR_DEFAULT;
+            mi[i].ti.bg = COLOR_DEFAULT;
+            mi[i].mid = MID_SpiroCorner;
+            mi[i].invoke = CVMenuPointType;
+            i++;
+            mi[i].ti.text = (unichar_t *) _("Left Constraint ([)");
+            mi[i].ti.text_is_1byte = true;
+            mi[i].ti.fg = COLOR_DEFAULT;
+            mi[i].ti.bg = COLOR_DEFAULT;
+            mi[i].mid = MID_SpiroLeft;
+            mi[i].invoke = CVMenuPointType;
+            i++;
+            mi[i].ti.text = (unichar_t *) _("Right Constraint (])");
```

Comment:
What's the bracket about? (Not implying it's wrong.)

Reply:
I debated adding it, but I think it makes it more obvious what point type is being talked about, since the icon looks like a bracket. I think it's a UI enhancement to throw it in, but can take it out if you don't like it.

Reply:
I think it's best to leave it out -- it might look to some users like some unknown form of keyboard acceleration.

Reply:
Would have also affected i18n, now that I think of it. But anyway ...

Reply:
Right, I've taken it out already. :-)